src/v/iceberg/CMakeLists.txt

```diff
@@ -0,0 +1,36 @@
+find_package(Avro)
+
+set(schemas_dir "${CMAKE_CURRENT_SOURCE_DIR}/avroschemas")
+
+set(schemas
+  manifest_file.schema.json
+  manifest_entry.schema.json
+)
+
+set(avro_hdrs)
+foreach (schema ${schemas})
+  get_filename_component(struct_name ${schema} NAME_WE)
+  set(avro_hdr "${CMAKE_CURRENT_BINARY_DIR}/${struct_name}.h")
+
+  add_custom_command (
+    OUTPUT ${avro_hdr}
+    COMMAND avrogencpp
+    -i ${schemas_dir}/${schema}
+    -o ${avro_hdr} -n iceberg
+    DEPENDS avrogencpp ${schemas_dir}/${schema}
+    COMMENT "Running avrogencpp on ${schema}"
+    VERBATIM)
+
+  list(APPEND avro_hdrs ${avro_hdr})
+endforeach()
+
+add_custom_target(
+  iceberg_avro_codegen
+  DEPENDS ${avro_hdrs})
+
+v_cc_library(
+  NAME iceberg
+)
+add_dependencies(v_iceberg iceberg_avro_codegen)
```

Comment:
instead of `add_dependencies`, add the headers as source files to the iceberg v_cc_library like `SRCS ${avro_hdrs}` you can then remove the custom target too.

Reply:
I had this originally, but I don't think it works with no other .cc files (at least, when I do this, the open source build fails)

Reply:
how about this

```
commit b92f9af5868692e5794f050a73b141fa0c6c630b (HEAD -> iceberg-avro)
Author: Noah Watkins <noahwatkins@gmail.com>
Date:   Mon Jul 15 13:13:35 2024 -0700

    fix

    Signed-off-by: Noah Watkins <noahwatkins@gmail.com>

diff --git a/src/v/iceberg/CMakeLists.txt b/src/v/iceberg/CMakeLists.txt
index c89b07563c..6eb193ab3f 100644
--- a/src/v/iceberg/CMakeLists.txt
+++ b/src/v/iceberg/CMakeLists.txt
@@ -24,13 +24,9 @@ foreach (schema ${schemas})
   list(APPEND avro_hdrs ${avro_hdr})
 endforeach()

-add_custom_target(
-  iceberg_avro_codegen
-  DEPENDS ${avro_hdrs})
-
 v_cc_library(
   NAME iceberg
+  SRCS ${avro_hdrs}
 )
-add_dependencies(v_iceberg iceberg_avro_codegen)

 add_subdirectory(tests)

commit e497ea1036d2008272f7db6459fa913d3dbdb664
Author: Noah Watkins <noahwatkins@gmail.com>
Date:   Mon Jul 15 13:10:06 2024 -0700

    cmake: populate interface target sources

    Signed-off-by: Noah Watkins <noahwatkins@gmail.com>

diff --git a/cmake/v_library.cmake b/cmake/v_library.cmake
index 40bb1e9301..544bd07e9a 100644
--- a/cmake/v_library.cmake
+++ b/cmake/v_library.cmake
@@ -106,7 +106,7 @@ function(v_cc_library)
       )
   else()
     # Generating header-only library
-    add_library(${_NAME} INTERFACE)
+    add_library(${_NAME} INTERFACE ${V_CC_LIB_SRCS})
     target_include_directories(${_NAME}
       INTERFACE ${CMAKE_CURRENT_LIST_DIR}/include)
     target_link_libraries(${_NAME}
```

Reply:
That works! Thanks.

Done